src/_main_/utils/translation/__init__.py

```diff
@@ -45,11 +71,15 @@ def _should_exclude(self, key, _value):
         # Check if key is in exclude_keys or JSON_EXCLUDE_KEYS
         if key in self.exclude_keys or key in JSON_EXCLUDE_KEYS:
             return True
-        
+
         # Check if value is not a string.  For eg. we want to exclude types like bool, int etc
-        if not isinstance(_value, str) :
+        if not isinstance(_value, str):
+            return True
+
+        # Check if value is in JSON_EXCLUDE_VALUES
+        if _value in JSON_EXCLUDE_VALUES or any(re.fullmatch(pattern, _value) for pattern in EXCLUDED_JSON_VALUE_PATTERNS):
```

Comment:
i'm ok with this for now but i am curious about the impact on the runtime.  for large text blocks does the speed decrease because of the regex comparison?

Reply:
I will do some reading and run some experiments with fairly large bodies of text to see how it fare and revert

Reply:
From my readings on similar issues and a Stackoverflow, [here](https://stackoverflow.com/questions/42742810/speed-up-millions-of-regex-replacements-in-python-3), pre-compiling the regex patterns dramatically improves the performance and good for our use-case.